internal/cli/global.go

```diff
@@ -51,19 +51,22 @@ func (o *GlobalOptions) Validate(args []string) error {
 		return fmt.Errorf("request-timeout must be greater than 0")
 	}
 
-	if o.ConfigDir != "" {
-		path := filepath.Clean(o.ConfigDir)
-		ext := filepath.Ext(path)
-		if ext != "" {
-			return fmt.Errorf("config-dir should specify a directory path")
-		}
+	// Validate ConfigDir exists and is a directory
+	if err := validateConfigDir(filepath.Dir(o.ConfigFilePath)); err != nil {
+		return err
 	}
 
-	if _, err := os.Stat(o.ConfigFilePath); errors.Is(err, os.ErrNotExist) {
-		if o.Context != "" {
-			return fmt.Errorf("context '%s' does not exist", o.Context)
+	// Validate ConfigFilePath is not a directory and exists as a file
+	if stat, err := os.Stat(o.ConfigFilePath); err != nil {
```

Comment:
Why not just call `validateConfigDir` on `o.ConfigFilePath` above instead of adding this check? This is duplicating functionality